### PR TITLE
Rename role and simplify

### PR DESCRIPTION
**Pull request by spring1843**
**1. Issue, if available:**

**2. Description of changes:**
* s/nightly/Nightly
* Removes extra lines from nightly.sh

**3. How was this change tested?**

**4. Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: hack/nightly.sh [hunk 1]
```diff
@@ -4,19 +4,6 @@ RELEASE_REPO=${RELEASE_REPO:-public.ecr.aws/z3c6l6z9/karpenter-nightly}
 RELEASE_VERSION=${RELEASE_VERSION:-$(date "${NIGHTLY_TAG_FMT}")}
 RELEASE_PLATFORM="--platform=linux/amd64,linux/arm64"
 
-if [ -z "$CLOUD_PROVIDER" ]; then
-    echo "CLOUD_PROVIDER environment variable is not set: 'export CLOUD_PROVIDER=aws'"
-    exit 1
-fi
```
Comment (njtran) [inline hack/nightly.sh:10]: Why do we want to remove this? We'd want to know if no cloud provider is set so we don't silently default to a mock environment.

File: hack/nightly.sh [hunk 2]
```diff
@@ -4,19 +4,6 @@ RELEASE_REPO=${RELEASE_REPO:-public.ecr.aws/z3c6l6z9/karpenter-nightly}
...
-
-# TODO restore https://reproducible-builds.org/docs/source-date-epoch/
```
Comment (njtran) [inline hack/nightly.sh:12]: Are you removing this because it's been completed?

Comment (spring1843) [reply hack/nightly.sh:12]: No those were copied from release.sh where there's cosign they are not used in here.

File: hack/nightly.sh (hunk 1 above)
Comment (spring1843) [reply hack/nightly.sh:10]: this was removed from https://github.com/aws/karpenter/pull/1587 and it's not needed in the nightly as there's no mock env.

Comment (bwagner5) [reply hack/nightly.sh:10]: the same thing could happen in the nightly release. If the env var isn't set, Karpenter will build with our "fake" cloud provider. IMO this should be added back to the nightly script.

Comment (spring1843) [reply hack/nightly.sh]: Alright added back.

**Review by bwagner5 — APPROVED** [review]
lgtm!

**Pull request merged**